Fix/full invert crash 2942

Abdurrahman-shaikh opened the pull request:
# Description of Changes

Please provide a summary of the changes, including:

- What was changed
  - Modified the `convertToBufferedImageTpFile` to use `File.createTempFile()` instead of writing to `"image.png"` in the current directory.
   - This change ensures the file is saved in the default temporary directory, preventing permission issues.

- Why the change was made
  - Previously, the method attempted to save the file in the current working directory, which caused permission errors (`java.io.FileNotFoundException: image.png (Permission denied)`).
 
- Any challenges encountered

Closes #2942

---

## Checklist

### General

- [x] I have read the [Contribution Guidelines](https://github.com/Stirling-Tools/Stirling-PDF/blob/main/CONTRIBUTING.md)
- [x] I have read the [Stirling-PDF Developer Guide](https://github.com/Stirling-Tools/Stirling-PDF/blob/main/DeveloperGuide.md) (if applicable)
- [ ] I have read the [How to add new languages to Stirling-PDF](https://github.com/Stirling-Tools/Stirling-PDF/blob/main/HowToAddNewLanguage.md) (if applicable)
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings

### Documentation

- [ ] I have updated relevant docs on [Stirling-PDF's doc repo](https://github.com/Stirling-Tools/Stirling-Tools.github.io/blob/main/docs/) (if functionality has heavily changed)
- [ ] I have read the section [Add New Translation Tags](https://github.com/Stirling-Tools/Stirling-PDF/blob/main/HowToAddNewLanguage.md#add-new-translation-tags) (for new translation tags only)

### UI Changes (if applicable)

- [ ] Screenshots or videos demonstrating the UI changes are attached (e.g., as comments or direct attachments in the PR)

### Testing (if applicable)

- [x] I have tested my changes locally. Refer to the [Testing Guide](https://github.com/Stirling-Tools/Stirling-PDF/blob/main/DeveloperGuide.md#6-testing) for more details.

Frooodle commented on src/main/java/stirling/software/SPDF/utils/misc/InvertFullColorStrategy.java:
```diff
@@ -98,8 +98,10 @@ private void invertImageColors(BufferedImage image) {
+        File file = File.createTempFile("image", ".png");
         ImageIO.write(image, "png", file);
+        //Ensures temp file is deleted when jvm terminates
+        file.deleteOnExit();
```
Delete on exit isn't enough since this server could be running for weeks or months, we must delete it after operation
Can you replace this with a try finally?

Abdurrahman-shaikh commented:
@Frooodle I’ve updated the code to address your feedback. Temporary files are now cleaned up immediately using `try-finally` blocks. Could you please review the changes?
